src/current/v25.2/row-level-security.md

```diff
@@ -0,0 +1,569 @@
+~~~
+
+#### Define user roles for app developer
+
+{% include_cached copy-clipboard.html %}
+~~~ sql
+CREATE ROLE app_dev;
+GRANT SELECT ON tenants TO app_dev;
+GRANT SELECT, INSERT, UPDATE, DELETE ON invoices TO app_dev;
+~~~
+
+#### Define how the application sets the tenant ID for the session
+
+Next, each application will need to set the tenant context for the session. In this example, we will use the `application_name` session variable to pass in a tenant ID which will later be extracted from the variable.
+
+Specifically, the UUID following the period in `application_name` is the tenant ID. We will use the `current_setting()` function in our RLS policies to extract the ID.
+
+{{site.data.alerts.callout_danger}}
+For multi-tenancy to work correctly, this setting **must** be reliably managed by the application layer and passed in the connection string.
+{{site.data.alerts.end}}
+
+{% include_cached copy-clipboard.html %}
+~~~ sql
+SET application_name = 'my_cool_app.9607a12c-3c2f-407b-ae3c-af903542395b';
+~~~
+
+#### Enable row-level security for multi-tenant isolation
+
+To enable row-level security for the `invoices` table, issue the statements below.
+
+{{site.data.alerts.callout_danger}}
+For multi-tenant isolation to work properly in this example, you **must** also `FORCE ROW LEVEL SECURITY` so that the policies also apply to the table owner.
+{{site.data.alerts.end}}
+
+{% include_cached copy-clipboard.html %}
+~~~ sql
+ALTER TABLE invoices ENABLE ROW LEVEL SECURITY, FORCE ROW LEVEL SECURITY;
+~~~
+
+#### Define tenant isolation policies
+
+The following policy enforces tenant isolation for all operations.
+
+{% include_cached copy-clipboard.html %}
+~~~ sql
+CREATE POLICY tenant_isolation_permissive ON invoices
+    AS PERMISSIVE
+    FOR ALL
+    TO public
+    USING (tenant_id = COALESCE(
+        -- Attempt to extract UUID after the '.' in application_name
+        -- Handle cases where setting is missing or improperly formatted
+        NULLIF(split_part(current_setting('application_name', true), '.', 2), '')::UUID,
+        -- If no tenant ID is successfully extracted, use the Default Tenant ID
+        'ad7b4529-f5d6-4d0e-8b0f-91af455a70e5'
+        )
+    );
+
+CREATE POLICY tenant_isolation ON invoices
+    AS RESTRICTIVE
+    FOR ALL
+    TO app_dev
+    USING (tenant_id = split_part(current_setting('application_name', true),'.',2)::UUID) -- Filter rows on SELECT/UPDATE/DELETE
+    WITH CHECK (tenant_id = split_part(current_setting('application_name', true),'.',2)::UUID); -- Enforce tenant_id on INSERT/UPDATE
+~~~
+
+Explanation of policy:
+
+- `AS PERMISSIVE`: Necessary because you need at least one permissive policy. The permissive policy above has logic to show results for a default tenant ID if the `application_name` is omitted or improperly formatted.
+- `AS RESTRICTIVE`: Makes the policy mandatory. If other policies exist, they must *also* pass. For simple tenant isolation, this is often the safest default.
```

Comment:
I think this is a good example. We should mention that the restrictive policy applies to app_dev role. Meaning, anyone assigned to the app_dev role must use the correctly formatted application_name, and we don't want them to fallback to the default tenant.

Reply:
thanks, updated to add 

> - `AS RESTRICTIVE`: Makes the policy mandatory. If other policies exist, they must *also* pass. For simple tenant isolation, this is often the safest default. The restrictive policy above applies to the `app_dev` role, so that anyone assigned to the `app_dev` role must use the correctly formatted `application_name`, and is not allowed to fallback to the default tenant ID.